var/spack/repos/builtin.mock/packages/view-ignore-conflict/package.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+
+from spack.package import *
+
+
+class ViewIgnoreConflict(Package):
```

Comment:
(minor) FWIW: I started finding it cumbersome to define multiple `builtin.mock` packages so I created a mechanism to define them as strings in whatever test file needs them. See for example https://github.com/spack/spack/pull/41406/files#diff-3d650e694e939950596b60fd4356346e0109f0d8171b92720a75e3664952d44e

Reply:
I like that but would really like to just write

```python
def test_smth(create_package):
  class TestPackage(Package):
    def install(...):
      ...
  create_package(TestPackage)
```

which may or may not work with `inspect.getsource`.